slack-base/src/main/java/com/hubspot/slack/client/models/events/user/SlackUserChangeEventIF.java

```diff
@@ -0,0 +1,25 @@
+package com.hubspot.slack.client.models.events.user;
+
+import org.immutables.value.Value.Immutable;
+
+import com.fasterxml.jackson.databind.PropertyNamingStrategy.SnakeCaseStrategy;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import com.hubspot.immutables.style.HubSpotStyle;
+import com.hubspot.slack.client.models.events.SlackEvent;
+import com.hubspot.slack.client.models.users.SlackUser;
+
+@Immutable
+@HubSpotStyle
+@JsonNaming(SnakeCaseStrategy.class)
+@JsonDeserialize(as = SlackUserChangeEvent.class)
+public interface SlackUserChangeEventIF extends SlackEvent {
+
+  SlackUser getUser();
+
+  //User change events do not have a ts, so we manually set it as null
+  @Override
+  default String getTs() {
+    return null;
```

Comment:
I'm a little concerned about this. What you're really saying is that this is actually nullable, but by doing it this way you've avoided changing the method signature. This exposes the possibility of a NPE down the road, because you've violated the contract. The right answer is to make slack events have optional TS since not every slack event actually has a TS, or to make TS not be a required field on the base `SlackEvent`.

Reply:
Making getTs optional is a breaking change, but removing getTs from the `SlackEvent` so that it's not required sounds like it would achieve the desired effects and circumvent breaking existing usages. Then we can just define getTs in the events that have it. For this repo's purposes, it's just `SlackEventMessage` for `Message` events. I think that makes good sense

Reply:
See discussion on https://github.com/HubSpot/slack-client/issues/52